### PR TITLE
deprecation(semver): rename `isSemVerComparator()`

### PR DESCRIPTION
**Pull request by timreichen**
renames
`SemVerComparator` => `Comparator`
`isSemVerComparator()` => `isComparator()`
~~`isSemVerRange()` => `isRange()`~~
~~`SemVerRange` => `Range`~~
All old functions and interfaces are deprecated.

### CODE REVIEW

File: semver/is_semver_comparator_test.ts
Comment (iuioiua) [inline semver/is_semver_comparator_test.ts:1]: I think this file should be `is_comparator_test.ts`.

File: semver/is_semver_range.ts
Comment (iuioiua) [inline semver/is_semver_range.ts:14]:
```suggestion
 * @deprecated (will be removed in 0.212.0) Use {@linkcode isRange} instead.
```

File: semver/is_semver_comparator.ts
Comment (iuioiua) [inline semver/is_semver_comparator.ts:14]:
```suggestion
 * @deprecated (will be removed in 0.212.0) Use {@linkcode isComparator} instead.
```

File: semver/types.ts
Comment (iuioiua) [inline semver/types.ts:56]:
```suggestion
 * @deprecated (will be removed in 0.212.0) Use {@linkcode Comparator} instead.
```

**Review by iuioiua — CHANGES_REQUESTED** [review]

**Comment by timreichen** [comment]
@iuioiua Before merging: There is also an opportunity to simplify `Range` and lose the single `ranges` property:

before
```ts
export interface Range {
  // The outer array is OR while each inner array is AND
  ranges: SemVerRangeOr;
}
```
after
```ts
// The outer array is OR while each inner array is AND
export type Range = Comparator[][];
```
WDYT?

File: semver/is_semver_comparator_test.ts
Comment (timreichen) [reply semver/is_semver_comparator_test.ts:1]: Done

File: semver/is_semver_comparator.ts
Comment (iuioiua) [inline semver/is_semver_comparator.ts:18]:
```suggestion
export const isSemVerComparator = isComparator;
```
Sorry, should've spotted this earlier.

File: semver/is_semver_range.ts
Comment (iuioiua) [inline semver/is_semver_range.ts:18]:
```suggestion
export const isSemVerRange = isRange;
```

**Review by iuioiua — CHANGES_REQUESTED** [review]
Sorry, I missed a couple of nits.

**Review by iuioiua — CHANGES_REQUESTED** [review]
Sorry, I missed a couple of nits.

**Comment by timreichen** [comment]
No problem.

> @iuioiua Before merging: There is also an opportunity to simplify `Range` and lose the single `ranges` property:
> 
> before
> 
> ```ts
> export interface Range {
>   // The outer array is OR while each inner array is AND
>   ranges: SemVerRangeOr;
> }
> ```
> 
> after
> 
> ```ts
> // The outer array is OR while each inner array is AND
> export type Range = Comparator[][];
> ```
> 
What is your opinion on this?

**Comment by timreichen** [comment]
Actually, looking into Range a bit more, it seems that the current implementation is kinda strange.
It is defined like multiple segments on a line. Imo Range implies to be a single segment.

**Comment by iuioiua** [comment]
Yeah, it is strange. I see repeated use of `range.ranges`. It should probably be just `ranges` (without the `ranges` property).

**Comment by timreichen** [comment]
How about we redefine Range to `export type Range = Comparator[]` and remove one OR layer. The OR layer is based on `parseRange` implementation which can parse `range || range`, but  it should actually return Range[] instead of Range and be called `parseRanges` instead of `parseRange`.

**Comment by iuioiua** [comment]
The first solution sounds better.

**Comment by timreichen** [comment]
> Yeah, it is strange. I see repeated use of `range.ranges`. It should probably be just `ranges` (without the `ranges` property).

The more I investigate about the Range implementation, the more am I convinced that the two renames
- isSemVerRange() => isRange()
- SemVerRange => Range

are misleading.

Their actual definition of SemVerRange is not a single Range but an array of Ranges.
Instead of renaming Ranges, we should implement `Range` and also adjust the API functions to accept Range instead of Ranges. Basically all range* functions (`rangeFormat`, `rangeMin`, `rangeMax`, `rangeIntersects`, `minSatisfying`, `maxSatisfying`,  `testRange`, etc.) just loop over the Range array anyway. So instead I would suggest changing these to accept a single Range and get rid of all these loops. WDYT?

**Please do not merge** until this fundamental problem is solved.

**Comment by iuioiua** [comment]
Are you able to open a new issue focusing on this weird nested range design? There may be a reason (or there may not be) for this design decision.

Either way, let's proceed with this PR as it focuses on removing the `SemVer` portion of these symbol names.

**Comment by timreichen** [comment]
@iuioiua I will open a new issue for this. This misnomer is also present in the npm semver.  Even the bnf file https://github.com/npm/node-semver/blob/main/range.bnf calls it `range-set`.

Would you be ok if I reduced this PR to
- `isSemVerComparator()` => `isComparator()`
- `SemVerComparator` => `Comparator`

so it can be merged?
Else we potentially would need to re-deprecate the `Range` later again and wait a cycle until we can fix the problem.

**Comment by iuioiua** [comment]
Yep, let's reduce the PR 👍🏾

File: semver/is_semver_comparator.ts
Comment (timreichen) [reply semver/is_semver_comparator.ts]: done

**Comment by timreichen** [comment]
> Yep, let's reduce the PR 👍🏾

done

**Review by iuioiua — APPROVED** [review]
LGTM! Thanks! Let's see what Yoshiya thinks.

**Review by kt3k — APPROVED** [review]
LGTM

**Pull request merged**
